DeploymentAgentReconfigurationScripts/TFS2018 Update2/HardwareMoveRemoveOldAgent.ps1

```diff
@@ -0,0 +1,178 @@
+        $existingAgentFolder = $serviceExePath.Substring(1, $serviceExePath.Length - 22);
+    }
+}
+
+if ($existingAgentFolder -eq ""){
+    throw "Not able to auto detect existing agent folder. Provide the existingAgentFolder as input parameter. You can find it in agent capabilities UI as Agent.HomeDirectory.  Generally it is in C:\vstsagent folder.";
+}
+
+cd $existingAgentFolder;
+
+if (-not (Test-Path '.\.agent')){
+    throw "No agent installed in this path. Please run this script from Agent Home Directory. Generally it is in C:\vstsagent folder."
+}
+
+
+# Collect information about the existing agent
+$sourceTFSUrl = "NoAbleToReadAgentFile"
+$collectionName = "NoAbleToReadAgentFile"
+$projectName = "NoAbleToReadAgentFile"
+$deploymentGroupId = "NoAbleToReadAgentFile"
+$deploymentGroupName = "NoAbleToReadAgentFile"
+$agentName = "NoAbleToReadAgentFile"
+$tags = "NoAbleToReadAgentFile"
+$deploymentMachineId = "NoAbleToReadAgentFile"
+
+foreach($line in Get-Content .\.agent) {
+    $token = $line.split('"');
+
+    if ($token[1] -eq 'serverUrl'){
+        $sourceTFSUrl = $token[3];
+        if ($sourceTFSUrl.StartsWith($targetTFSUrl)){
+            Write-Verbose -Verbose "Agent is already configured to the TFS $targetTFSUrl";
+            return 0;
+        }
+    }
+
+    if ($token[1] -eq 'collectionName'){
+        $collectionName = $token[3];
+    }
+
+    if ($token[1] -eq 'projectId'){
+        $projectName = $token[3];
+    }
+
+    if ($token[1] -eq 'deploymentGroupId'){
+        $deploymentGroupId = $token[2].Substring(2, $token[2].Length - 3);
+    }
+
+    if ($token[1] -eq 'agentName'){
+        $agentName = $token[3];
+    }
+}
+
+# Get the Deployment group name
+$encodedPat = [System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes(":$patToken"))
+$getDeploymentGroupUrl = $targetTFSUrl + '/' + $collectionName + '/' + $projectName + '/_apis/distributedtask/deploymentgroups/' + $deploymentGroupId + '?api-version=5.0-preview.1'
+ 
+$dg = Invoke-RestMethod -Uri $getDeploymentGroupUrl -Method Get -Headers @{Authorization = "Basic $encodedPat"} -ContentType "application/json"
+$deploymentGroupName = $dg.name
+
+
+# Get the Machine Tags
+$getDeploymentMachineUrl = $targetTFSUrl + '/' + $collectionName + '/' + $projectName + '/_apis/distributedtask/deploymentgroups/' + $deploymentGroupId + '/targets?name=' + $agentName + '&api-version=5.0-preview.1'
+
+$dm = Invoke-RestMethod -Uri $getDeploymentMachineUrl -Method Get -Headers @{Authorization = "Basic $encodedPat"} -ContentType "application/json"
+$tags = $dm.value[0].tags -join ",";
+$deploymentMachineId = $dm.value[0].id;
+
+if ($dm.value[0].agent.status -ne 'offline'){
+    Write-Verbose -Verbose "Agent is already configured to the TFS $targetTFSUrl"
+    return 0;
+}
+
+
+# Create and go to new agent folder
+If(-NOT (Test-Path $env:SystemDrive\'vstsagent'))
+{
+    mkdir $env:SystemDrive\'vstsagent'
```

Comment:
If the user has existing agent in a custom path, should we create new agents also in the same path? you any way have that path info already.

Reply:
Sure. Will do that